velox/dwio/parquet/tests/reader/ParquetReaderExample.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <folly/init/Init.h>
+#include <cstdint>
+#include <iostream>
+#include "velox/dwio/parquet/reader/ParquetReader.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::common;
+using namespace facebook::velox::dwio::common;
+using namespace facebook::velox::parquet;
+
+std::shared_ptr<facebook::velox::common::ScanSpec> makeScanSpec(
+    const RowTypePtr& rowType) {
+  auto scanSpec = std::make_shared<facebook::velox::common::ScanSpec>("");
+  scanSpec->addAllChildFields(*rowType);
+  return scanSpec;
+}
+
+// A temporary program that reads from parquet file and prints its meta data and
+// content. Usage: velox_scan_parquet {parquet_file_path}
+int main(int argc, char** argv) {
+  folly::init(&argc, &argv, false);
+
+  if (argc < 2) {
+    std::cerr << "Usage: " << argv[0] << " <parquet file>" << std::endl;
+    return 1;
+  }
+  std::string filePath{argv[1]};
+
+  auto pool = facebook::velox::memory::addDefaultLeafMemoryPool();
+
+  facebook::velox::dwio::common::ReaderOptions readerOpts{pool.get()};
+  auto reader = new ParquetReader(
+      std::make_unique<BufferedInput>(
+          std::make_shared<LocalReadFile>(filePath),
+          readerOpts.getMemoryPool()),
+      readerOpts);
+
+  std::cout << "number of rows: " << reader->numberOfRows().value()
+            << std::endl;
+
+  auto outputType = reader->rowType();
+  std::cout << "velox type: " << outputType->toString() << std::endl;
```

Comment:
It would be great to print the encoding type of each page. In the future we can improve the program to specify the options whether to print them or not, and also specify which page we want to print.

Reply:
Are you looking for output like `parquet-tools`?
```
lniu@devrestricted-lniu:~$ parquet-tools inspect /home/lniu/parquet_test/testArray/test_1.parquet

############ file meta data ############
created_by: parquet-mr-trino version test-version (build n/a)
num_columns: 1
num_rows: 17900
num_row_groups: 1
format_version: 1.0
serialized_size: 285


############ Columns ############
element

############ Column(element) ############
name: element
path: test.list.element
max_definition_level: 3
max_repetition_level: 1
physical_type: INT32
logical_type: Int(bitWidth=32, isSigned=true)
converted_type (legacy): INT_32
compression: GZIP (space_saved: 48%)
```
Yeah it will be useful to show those info, but the short term goal for this program is to use it to test velox parquet reader. I think we can open another issue as a stretch goal.

Reply:
> Are you looking for output like parquet-tools?

Yes. And we should also try to make the command line arguments similar to it. This is useful since users won't need to install parquet-tools separately anymore.